Lib/asyncio/tasks.py

```diff
@@ -720,7 +720,7 @@ async def sleep(delay, result=None):
         h.cancel()
 
 
-def ensure_future(coro_or_future, *, loop=None):
+def ensure_future(coro_or_future, *, loop=None, context=None):
```

Comment:
Sadly this change is a no-go.

If `coro_or_future` is a coroutine, the `context` argument would be ignored.

I suggest a different approach to this PR:

`Runner.run(o)` should check if `o` is a coroutine -- if it is, keep things as is. If not, it should wrap it in an ad-hoc "proxy" coroutine `async def proxy(o=o): return await o` and pass `proxy` to `create_task()`.

Reply:
I have made the requested changes; please review again.

Reply:
(apologies Yury for misspelling your first name in the last commit!)